src/pedsilicoICH/lesion_definition.py

```
@@ -55,26 +66,33 @@ def insert_dural_3D(phantom, init_slice, hematoma_type, mass_effect,
-            start_point = dura_idx[random.choice(range(len(dura_idx)))]
-            # print(start_point)
+            tol = 2000
+            count = 0
+            while count < tol:
+                temp_boundary = boundary[init_slice]
+                dura_idx = np.argwhere(temp_boundary == 1.0)
 
-            distance_idx = np.zeros(len(dura_idx))
-            for i in range(len(dura_idx)):
-                distance_idx[i] = math.sqrt((start_point[0] - dura_idx[i][0])**2 + (start_point[1] - dura_idx[i][1])**2)
+                try:
+                    # choose a random start point, and calculate distance from all available boundary voxels to start point
+                    start_point = dura_idx[random.choice(range(len(dura_idx)))]
+
+                    distance_idx = np.zeros(len(dura_idx))
+                    for i in range(len(dura_idx)):
+                        distance_idx[i] = math.sqrt((start_point[0] - dura_idx[i][0])**2 + (start_point[1] - dura_idx[i][1])**2)
 
-            # create list of possible end points and choose one at random
-            close_voxel_list = np.where(np.logical_and(distance_idx > distances[0], distance_idx < distances[1]))
-            end_point = dura_idx[random.choice(close_voxel_list[0])]
+                    # create list of possible end points and choose one at random
+                    close_voxel_list = np.where(np.logical_and(distance_idx > distances[0], distance_idx < distances[1]))
+                    end_point = dura_idx[random.choice(close_voxel_list[0])]
 
-            orig_start = start_point
-            orig_end = end_point
+                    orig_start = start_point
+                    orig_end = end_point
+                except:
+                    count += 1
+                    init_slice = int(random.choice(np.linspace(0, int(HU_array.shape[0]/3), int(HU_array.shape[0]/3) + 1)))
+                else:
+                    count = tol
```

it would be useful to add in an Error message if this while loop is failing due to requesting too large of a volume. As currently implemented a try-except clause means that any errors are hidden which can make it hard for a user to realize they are requesting too large of a volume.